### PR TITLE
Workaround OneCycleLR state_dict bug

### PR DESCRIPTION
**Pull request by mrcslws**
Workaround for https://github.com/pytorch/pytorch/issues/42376. Avoids running into this issue in future checkpoints. Adds a script for repairing existing checkpoints.

### CODE REVIEW

File: nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py [hunk 1]
```diff
@@ -719,7 +719,12 @@ def get_state(self):
 
         if self.lr_scheduler is not None:
             with io.BytesIO() as buffer:
-                serialize_state_dict(buffer, self.lr_scheduler.state_dict())
+                state_dict = self.lr_scheduler.state_dict()
+                if "anneal_func" in state_dict:
+                    # FIXME: This is a workaround for a PyTorch bug.
+                    # https://github.com/pytorch/pytorch/issues/42376
+                    del state_dict["anneal_func"]
```
Comment (lscheinkman) [inline nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py:726]: You should store the `anneal_strategy` instead and restore the correct function on load.

File: projects/imagenet/repair_onecyclelr_checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    fix_needed = False
+    if "lr_scheduler" in checkpoint:
+        print("Loading LR scheduler state dict (this might take a few minutes)")
+        with io.BytesIO(checkpoint["lr_scheduler"]) as buf:
+            lr_sched_state_dict = deserialize_state_dict(buf)
+
+        if "anneal_func" in lr_sched_state_dict:
+            fix_needed = True
+            del lr_sched_state_dict["anneal_func"]
```
Comment (lscheinkman) [inline projects/imagenet/repair_onecyclelr_checkpoint.py:49]: You should store the `anneal_strategy` instead and restore the correct function on load.

File: nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py (hunk 1 above)
Comment (mrcslws) [reply nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py:726]: I think that would be overkill. The state dict is used for time-varying attributes. There is no supported way to change the anneal strategy after the LR scheduler is instantiated. They include it in the state dict because they're lazily saving the whole `self.__dict__` rather than writing proper `state_dict` methods.

File: projects/imagenet/repair_onecyclelr_checkpoint.py (hunk 2 above)
Comment (mrcslws) [reply projects/imagenet/repair_onecyclelr_checkpoint.py:49]: I think this would be overkill. (see above)

Comment (lscheinkman) [reply projects/imagenet/repair_onecyclelr_checkpoint.py:49]: makes sense
